src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -275,6 +278,18 @@ private Query getRangeQuery(RangeQuery rangeQuery, IndexState state) {
     return ((RangeQueryable) field).getRangeQuery(rangeQuery);
   }
 
+  private Query getGeoBoundingBoxQuery(GeoBoundingBoxQuery geoBoundingBoxQuery, IndexState state) {
+    String fieldName = geoBoundingBoxQuery.getField();
+    FieldDef field = state.getField(fieldName);
+
+    if (!(field instanceof LatLonFieldDef)) {
+      throw new IllegalArgumentException(
+          "Field: " + fieldName + " does not support GeoBoundingBoxQuery");
+    }
```

Comment:
Could we maybe do this by creating a GeoQueryable field property instead. With the pluggable nature of field types, it is better to not tie ourselves to specific class implementations. Also, I imagine we will want to add geo distance and polygon query support in the future.

Reply:
I thought about doing it initially but thought it might be early to add it. But it does make it more cleaner.